src/mlpack/methods/ann/augmented/tasks/sort_impl.hpp

```diff
@@ -0,0 +1,69 @@
+/**
+ * @file sort_impl.hpp
+ * @author Konstantin Sidorov
+ *
+ * Implementation of SortTask class
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_AUGMENTED_TASKS_SORT_IMPL_HPP
+#define MLPACK_METHODS_AUGMENTED_TASKS_SORT_IMPL_HPP
+
+#include "sort.hpp"
+
+#include <cassert>
+#include <vector>
+#include <algorithm>
+#include <utility>
+
+using std::vector;
+using std::pair;
+using std::make_pair;
```

Comment:
I know it can make your code longer but `using` declarations inside of a header file are a bad idea; they apply to user code also, which can be problematic in some cases.